signalfx-java/src/main/java/com/signalfx/connection/AbstractHttpReceiverConnection.java

```diff
@@ -38,11 +38,12 @@ public abstract class AbstractHttpReceiverConnection {
     protected final HttpHost host;
     protected final RequestConfig requestConfig;
 
-    protected AbstractHttpReceiverConnection(SignalFxReceiverEndpoint endpoint, int timeoutMs,
+    protected AbstractHttpReceiverConnection(SignalFxReceiverEndpoint endpoint, int timeoutMs, int maxRetries,
```

Comment:
We definitely want to keep the existing constructor so that existing implementations don't break. How about an overloaded constructor that defaults the `maxRetries` to 3 and calls the other constructor? That should be more backwards compatible that way.